enterprise/views.py

```diff
@@ -0,0 +1,184 @@
+        """
+        request.session['sso_quarantined_modules'] = (
+            'enterprise.views',
+        )
+
+    @staticmethod
+    def lift_quarantine(request):
+        """
+        Remove the quarantine session variable.
+        """
+        request.session.pop('sso_quarantined_modules', None)
+
+    @staticmethod
+    def get_warning(provider, platform, required):
+        """
+        Get the appropriate warning for the form.
+        """
+        if required:
+            return _(
+                "Are you sure?\nIf you do not agree to share your data, you will have to use "
+                "another account to access {platform}."
+            ).format(platform=platform)
+        else:
+            return _(
+                "Are you sure?\nIf you do not agree to share your data, you will not get any "
+                "discounts from {provider}."
+            ).format(provider=provider)
+
+    @staticmethod
+    def get_note(provider, required):
+        """
+        Get the appropriate note for the form.
+        """
+        if required:
+            return _(
+                "*{provider} requires data sharing consent; if consent is not provided, you will"
+                " be redirected to log in page."
+            ).format(provider=provider)
+        else:
+            return _(
+                "*{provider} requests data sharing consent; if consent is not provided, you will"
+                " not be able to get any discounts from {provider}."
+            ).format(provider=provider)
+
+    def get(self, request):
+        """
+        Render a form to collect user input about data sharing consent.
+        """
+        self.patch_mako_lookup()
+
+        # Get the OpenEdX platform name
+        if configuration_helpers is None:
+            raise NotConnectedToEdX(_('Must be installed in an OpenEdX environment to use configuration helpers.'))
+        platform_name = configuration_helpers.get_value("PLATFORM_NAME", settings.PLATFORM_NAME)
+
+        # Get the EnterpriseCustomer for the request; raise an error if there isn't one.
+        customer = get_enterprise_customer_for_request(request)
+        if customer is None:
+            raise Http404
+
+        # Quarantine the user to this module.
+        self.quarantine(request)
+
+        required = customer.enforces_data_sharing_consent(EnterpriseCustomer.AT_LOGIN)
+
+        data = {
+            'platform_name': platform_name,
+            'sso_provider': customer.name,
+            'data_sharing_consent': 'required' if required else 'optional',
+            "messages": {
+                "warning": self.get_warning(customer.name, platform_name, required),
+                "note": self.get_note(customer.name, required),
+            },
+        }
+        if render_to_response is None:
+            raise NotConnectedToEdX(_('Must be installed in an OpenEdX environment to render this response.'))
```

Comment:
Same here - unnecessary, or should be. Please DRY this code by removing this repeated `if blah is None: raise NotConnectedToEdX` code throughout this file and just make sure that this whole module is only loaded/enabled (added to the urlconf) within the LMS.